plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/TimestampCoercer.java

```diff
@@ -83,4 +89,64 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
                             toType));
         }
     }
+
+    public static class VarcharToShortTimestampCoercer
+            extends TypeCoercer<VarcharType, TimestampType>
+    {
+        public VarcharToShortTimestampCoercer(VarcharType fromType, TimestampType toType)
+        {
+            super(fromType, toType);
```

Comment:
Might be useful check argument that TimestampType is short, given that we do `9 - toType.getPrecision()`